hazelcast-client/src/main/java/com/hazelcast/client/config/YamlClientConfigBuilder.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    public YamlClientConfigBuilder(InputStream in) {
+        this.in = in;
+    }
+
+    /**
+     * Loads the client config using the following resolution mechanism:
+     * <ol>
+     * <li>first it checks if a system property 'hazelcast.client.config' is set. If it exist and it begins with
+     * 'classpath:', then a classpath resource is loaded. Else it will assume it is a file reference</li>
+     * <li>it checks if a hazelcast-client.Yaml is available in the working dir</li>
+     * <li>it checks if a hazelcast-client.Yaml is available on the classpath</li>
+     * <li>it loads the hazelcast-client-default.Yaml</li>
+     * </ol>
+     */
+    public YamlClientConfigBuilder() {
+        this((YamlClientConfigLocator) null);
+    }
+
+    /**
+     * Constructs a {@link YamlClientConfigBuilder} that loads the configuration
+     * with the provided {@link YamlClientConfigLocator}.
+     * <p/>
+     * If the provided {@link YamlClientConfigLocator} is {@code null}, a new
+     * instance is created and the config is located in every possible
+     * places. For these places, please see {@link YamlClientConfigLocator}.
+     * <p/>
+     * If the provided {@link YamlClientConfigLocator} is not {@code null}, it
+     * is expected that it already located the configuration YAML to load
+     * from. No further attempt to locate the configuration YAML is made
+     * if the configuration YAML is not located already.
+     *
+     * @param locator the configured locator to use
+     */
+    public YamlClientConfigBuilder(YamlClientConfigLocator locator) {
+        if (locator == null) {
+            locator = new YamlClientConfigLocator();
+            locator.locateEverywhere();
+        }
+
+        this.in = locator.getIn();
+    }
+
+    public ClientConfig build() {
+        return build(Thread.currentThread().getContextClassLoader());
+    }
+
+    public ClientConfig build(ClassLoader classLoader) {
+        ClientConfig clientConfig = new ClientConfig();
+        build(clientConfig, classLoader);
+        return clientConfig;
+    }
+
+    public void setProperties(Properties properties) {
+        setPropertiesInternal(properties);
+    }
+
+    void build(ClientConfig clientConfig, ClassLoader classLoader) {
+        clientConfig.setClassLoader(classLoader);
+        try {
+            parseAndBuildConfig(clientConfig);
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        } finally {
+            IOUtil.closeResource(in);
+        }
+    }
+
+    private void parseAndBuildConfig(ClientConfig config) throws Exception {
+        YamlMapping yamlRootNode;
+        try {
+            yamlRootNode = ((YamlMapping) YamlLoader.load(in));
+        } catch (Exception ex) {
+            throw new InvalidConfigurationException("Invalid YAML configuration", ex);
+        }
+
+        YamlNode clientRoot = yamlRootNode.childAsMapping(ClientConfigSections.HAZELCAST_CLIENT.name.toLowerCase());
```

Comment:
Minor: `toLowerCase` is not really needed here.

Reply:
Removed.

Reply:
Now removed for real :)